File: creational/builder/python/director.py
from builder import PizzaBuilder


class PizzaDirector:
    def __init__(self, builder: PizzaBuilder):
        self._builder = builder

    def set_builder(self, builder: PizzaBuilder):
        self._builder = builder

    def construct_pizza(self, size):
        self._builder.set_size(size)
        self._builder.add_toppings()
        self._builder.set_crust_type()
        self._builder.set_sauce()
        self._builder.set_cheese()
        return self._builder.build()

File: creational/builder/typescript/director.ts
import { PizzaBuilder } from './builder';
import { Pizza } from './product';

export class PizzaDirector {
	constructor(private _builder: PizzaBuilder) {}

	setBuilder(builder: PizzaBuilder) {
		this._builder = builder;
	}

	constructPizza(size: string): Pizza {
		this._builder.setSize(size);
		this._builder.addToppings();
		this._builder.setCrustType();
		this._builder.setSauce();
		this._builder.setCheese();
		return this._builder.build();
	}
}
